### PR TITLE
tests: update say exercise

### PR DESCRIPTION
**Pull request by Elahi-cs**
This PR is the result of running this command:
image

using the [test creation helper tool](https://github.com/exercism/cpp/wiki/Test-Cases-for-Practice-Exercises#test-creation-helper-tool) linked in the wiki, with some modifications to the code necessary for the tool to work.

### CODE REVIEW

**Comment by Elahi-cs** [comment]
@vaeng everything should be fixed now. A couple of questions:
* I had to modify the exercise creation tool that you linked on the wiki, because it expects configlet to be on the same folder, but configlet expects to be run on the root folder of the repo. Do you see it necessary/useful to add these changes to the tool branch?
* Would it be useful to document the process to update these test cases? I'm talking setup tool -> run tool -> format with `clang-format` -> check for errors. I don't know if it's something that could be of help in the future or if it the process will be discarded once the 6 remaining exercises have their tests updated.

**Comment by vaeng** [comment]
Hello @Elahi-cs,

Thanks for your patience. You did the right thing to ask in the forum and also a nudge to look at this would have been fine. 

I think your change is good and welcome, I wanted to ask the other maintainers for feedback before going further, but I haven't had the time yet.

As a result of my late reply, this September or will become tagged for hacktoberfest.

I will not be able to check your code until Tuesday, but maybe @siebenschlaefer has some capacity this weekend.

File: exercises/practice/say/say_test.cpp [hunk 1]
```diff
@@ -5,93 +5,98 @@
...
-        say::in_english(987654321123ULL));
+        say::in_english(987654321123));
 }
 
-TEST_CASE("raises_an_error_below_zero")
-{
-    REQUIRE_THROWS_AS(say::in_english(-1), std::domain_error);
+TEST_CASE("numbers below zero are out of range",
+          "[49a6a17b-084e-423e-994d-a87c0ecc05ef]") {
+    REQUIRE("{'error': 'input out of range'}" == say::in_english(-1));
```
Comment (ahans) [inline exercises/practice/say/say_test.cpp:93]: It looks like the tool cannot handle cases with errors that well yet. For invalid input, we want the student to throw an exception.
```suggestion
    REQUIRE_THROWS_AS(say::in_english(-1), std::domain_error);
```

File: exercises/practice/say/say_test.cpp [hunk 2]
```diff
@@ -5,93 +5,98 @@
...
 }
 
-TEST_CASE("raises_an_error_for_one_trillion")
-{
-    REQUIRE_THROWS_AS(say::in_english(1000ULL * 1000ULL * 1000ULL * 1000ULL),
-                      std::domain_error);
+TEST_CASE("numbers above 999,999,999,999 are out of range",
+          "[4d6492eb-5853-4d16-9d34-b0f61b261fd9]") {
+    REQUIRE("{'error': 'input out of range'}" ==
+            say::in_english(1000000000000));
```
Comment (ahans) [inline exercises/practice/say/say_test.cpp:99]: As above. Fine to reformat the number, though:
```suggestion
    REQUIRE_THROWS_AS(say::in_english(1000000000000), std::domain_error);
```

**Review by ahans — CHANGES_REQUESTED** [review]
The changes look mostly fine, but the two cases giving an error in CI need to be fixed.

**Comment by ahans** [comment]
> Related question: is there a way to run the CI tests locally? I don't like making the pipeline run the tests every time, but I also don't know if there's a good way other than using `act`.

To run exactly what happens in CI, you probably won't get around using something such as `act`. But what happens is not exactly black magic and you can check the files in `.github` to see what's going on. The actual code tests are all managed via `CMake` and run [here](https://github.com/exercism/cpp/blob/main/.github/workflows/ci.yml#L24). So if you have all the prerequisites installed, you can this locally:

```sh
cmake .
# run all tests
cmake --build .
# alternatively, run just a single test:
cmake --build . -- test_say
```

**Review by vaeng — APPROVED** [review]
That looks good to me, I don't see any reason not to merge this now.
@ahans remarks seem to be addressed.

**Pull request merged**